### PR TITLE
Remove coverity from the DevGuide

### LINKED ISSUE #1410: Drop the info about Coverity Scans
Coverity Scans last scanned our repo in 2020.

### PR DESCRIPTION
**Pull request by Mariatta**
The CPython Coverity Scan hasn't been updated since 2020.
People listed as admins and experts of coverity scan are no longer administering/maintaining it.

On Discord, it was decided to drop this from CPython core workflow.

Closes #1410 

----
📚 Documentation preview 📚: https://cpython-devguide--1411.org.readthedocs.build/

### CODE REVIEW

File: development-tools/clang.rst [hunk 1]
```diff
@@ -54,7 +54,7 @@ A complete list of sanitizers can be found at `Controlling Code Generation
 
 Clang and its sanitizers have strengths (and weaknesses). Its just one tool in
 the war chest to uncovering bugs and improving code quality. Clang should be
-used to complement other methods, including Code Reviews, Valgrind, Coverity,
+used to complement other methods, including Code Reviews, Valgrind,
```
Comment (Mariatta) [inline development-tools/clang.rst:57]: Question, do we have more info about "Valgrind"? It was mentioned only once on the entire devguide.

Comment (Mariatta) [reply development-tools/clang.rst:57]: How to use valgrind? What is it for?

Comment (JelleZijlstra) [reply development-tools/clang.rst:57]: Valgrind is a tool for finding C-level memory issues, like memory leaks or writes to uninitialized memory.

Comment (encukou) [reply development-tools/clang.rst:57]: https://github.com/python/cpython/blob/main/Misc/README.valgrind

Comment (JelleZijlstra) [reply development-tools/clang.rst:57]: Interesting, we also have a https://github.com/python/cpython/blob/main/Misc/README.coverity that perhaps should be cleaned up from the repo.

Comment (Mariatta) [reply development-tools/clang.rst:57]: Yeah I found references to coverity on the CPython repo. In addition to README.coverity,
also `Misc\coverity_model.c`

Comment (Mariatta) [reply development-tools/clang.rst]: > It's documented here: https://github.com/python/cpython/blob/main/Misc/README.valgrind

I updated to link to that doc. Thanks.

File: development-tools/clang.rst [hunk 2]
```diff
@@ -281,3 +281,5 @@ Unfortunately, you won't know what to ignorelist until you run the sanitizer.
 
 The documentation is available at `Sanitizer special case list
 <https://clang.llvm.org/docs/SanitizerSpecialCaseList.html>`_.
+
+.. _Valgrind: https: https://github.com/python/cpython/blob/main/Misc/README.valgrind
```
Comment (JelleZijlstra) [inline development-tools/clang.rst:285]:
```suggestion
.. _Valgrind: https://github.com/python/cpython/blob/main/Misc/README.valgrind
```

**Review by JelleZijlstra — APPROVED** [review]

**Comment by Mariatta** [comment]
Related CPython PR: https://github.com/python/cpython/pull/124460

**Comment by Mariatta** [comment]
Thanks!

**Pull request merged**